### PR TITLE
338064 - Access more information by opening the 'Licence' tab.

### PR DESCRIPTION
**Pull request by karthigayayiniram**
### What one thing this PR does?
This task covers the License section of the tab-object presented on the NCEA Catalogue Details Page.

### Task details

- [338064 - Access more information by opening the 'Licence' tab](https://dev.azure.com/defragovuk/DEFRA-NCEA/_workitems/edit/338064/)

### How do these changes look like?

image
### Dev-tested in

1. Local environment

- [License](http://localhost:3000/search/af7cd14f-6c20-445f-abd0-7c41385ba999#license)
- 
2. Dev environment

- [License](http://localhost:3000/search/af7cd14f-6c20-445f-abd0-7c41385ba999#license)
- ...

### CODE REVIEW

**Review by ganesh-rajamahanthy — DISMISSED** [review]

File: src/utils/formatSearchResponse.ts [hunk 1]
```diff
@@ -73,13 +144,13 @@ const getOtherDetails = async (searchItem: Record<string, any>): Promise<Record<
...
     project_number: '',
     Metadata_language: searchItem?._source?.mainLanguage ?? '',
-    ncea_catalogue_date: catalogueDate
-      ? catalogueDate.toLocaleDateString('en-US', { day: 'numeric' }) +
-        '-' +
-        catalogueDate.toLocaleDateString('en-US', { month: 'short' }) +
-        '-' +
-        catalogueDate.toLocaleDateString('en-US', { year: 'numeric' })
-      : '',
+    ncea_catalogue_date: getCatalogueDate(searchItem),
```
Comment (FujaleAhmedShaik) [inline src/utils/formatSearchResponse.ts:147]: I believe we have done this to fix the function complexity suggested by Sonar. But we still have a few more tabs. After we add all the necessary things, this error might occur again. So maybe we can move this logic to another file and import it here.
With this, we can also write test scenarios for each tab of data that we receive and return.
Just for FYI, I have this feeling about the complexity issue and did similar things in my open [PR](https://github.com/DEFRA/ncea-frontend/blob/2dd07ec8d60836af5b77bd3f40a9b7f7f4a9571b/src/utils/formatSearchResponse.ts#L90).

File: src/utils/formatSearchResponse.ts [hunk 2]
```diff
@@ -54,15 +54,86 @@ const formatSearchResponse = async (
...
+  if (publishedBy.email) {
+    dataOwner += `${publishedBy.email}`;
+  }
+
+  return dataOwner;
+};
+
+const getCatalogueDate = (searchItem: Record<string, any>): string => {
   const catalogueDate = searchItem?._source?.dateStamp ? new Date(searchItem?._source?.dateStamp) : '';
+  return catalogueDate
```
Comment (FujaleAhmedShaik) [inline src/utils/formatSearchResponse.ts:105]: Can we not use the formatDate function that already exists and make the necessary changes without affecting the existing functionality?

File: src/utils/formatSearchResponse.ts [hunk 3]
```diff
@@ -54,15 +54,86 @@ const formatSearchResponse = async (
...
+  searchItem: Record<string, any>,
+  publishedBy: Record<string, any>,
+): Promise<Record<string, string>> => {
   return {
     alternateTitle: searchItem?._source?.resourceAltTitleObject?.[0]?.default ?? '',
     language: searchItem?._source?.mainLanguage?.toUpperCase() ?? '',
-    keywords: searchItem?._source?.tag?.map((item) => item.default).join(', ') ?? '',
-    topic_categories: searchItem?._source?.cl_topic?.map((item) => item.default).join(', ') ?? '',
+    keywords: getKeywords(searchItem),
+    topic_categories: getTopicCategories(searchItem),
```
Comment (FujaleAhmedShaik) [inline src/utils/formatSearchResponse.ts:134]: I have already made changes to this in my open [PR](https://github.com/DEFRA/ncea-frontend/pull/34).

**Review by FujaleAhmedShaik — CHANGES_REQUESTED** [review]

File: src/utils/formatSearchResponse.ts (hunk 2 above)
Comment (karthigayayiniram) [reply src/utils/formatSearchResponse.ts]: Addressed.

File: src/utils/formatSearchResponse.ts (hunk 3 above)
Comment (karthigayayiniram) [reply src/utils/formatSearchResponse.ts:123]: I had to do this because of the sonar error.

**Review by FujaleAhmedShaik — APPROVED** [review]

**Review by ganesh-rajamahanthy — APPROVED** [review]

**Pull request merged**